### ISSUE TITLE
Automatic emails for open access/subscription triggered post-install in OJS 3.0

### ISSUE THREAD

**Issue by mfelczak** [pkp/pkp-lib#1838]
At the completion of an OJS 3.0 install, the following emails are automatically sent out even though there's no subscription support:

> Subject: ##admin.scheduledTask.openAccessNotification## - 57d8449b0e2a3 - Error
> 
> Your Open Journal Systems installation automatically executed and finished this task and you can download the log file here: http://localhost:8111/index.php/index/admin/downloadScheduledTaskLogFile?file=%23%23admin.scheduledTask.openAccessNotification%23%23-57d8449b0e2a3-20160913.log

> Subject: Subscription expiry reminder - 57d8449acf9bb - Error
> 
> Your Open Journal Systems installation automatically executed and finished this task and you can download the log file here: http://localhost:8111/index.php/index/admin/downloadScheduledTaskLogFile?file=Subscriptionexpiryreminder-57d8449acf9bb-20160913.log

**Comment by ania-appb** [comment]
We have a similar problem.
After fresh installation of OJS 3.0 we get two notificatinos on a daily basis:

##admin.scheduledTask.openAccessNotification## - 57e39d5d6bdbb - Error

Subscription expiry reminder - 57e39d5cd73b0 - Error

Your Open Journal Systems installation automatically executed and finished this task and you can download the log file here...

Your ##common.openJournalSystems## installation automatically executed and finished this task and you can download the log file here


Why are we receiving them?  How to disable receiving these notifications?

Thank you, Anna

**Comment by ania-appb** [comment]
I posted this also in the forum:
http://forum.pkp.sfu.ca/t/subscription-renewal-reminders-being-sent-despite-policy-not-to-2-4-6/7418/5

And I would be really grateful for some advice how to, at least manually, disable these errant notifications, as they spam our mail.

Anna

**Comment by asmecher** [comment]
@mfelczak, is this something you'll have the chance to tinker with in the near future?

**Comment by ania-appb** [comment]
@mfelczak, Could I just remove the two first records from scheduled_tasks table in the database (see above)?

**Comment by mfelczak** [comment]
Hi @asmecher and @ania-appb, I'll have a PR (and patch) for this shortly.

**Comment by mfelczak** [comment]
@asmecher: PR at https://github.com/pkp/ojs/pull/1034

**Comment by mfelczak** [comment]
@ania-appb: you should be able to apply the above fix to your install and you'll need to manually delete the relevant rows from the scheduled_tasks table, i.e
```
DELETE FROM scheduled_tasks WHERE class_name = 'classes.tasks.OpenAccessNotification';
DELETE FROM scheduled_tasks WHERE class_name = 'classes.tasks.SubscriptionExpiryReminder';
```

**Comment by mfelczak** [comment]
@asmecher: updated PR at https://github.com/pkp/ojs/pull/1034

**Comment by ania-appb** [comment]
Thank you very much @mfelczak , half of the problem is resolved:
yesterday, I applied  fix #1838
 (I deleted lines 34-41 from  registry/scheduledTasks.xml)
 and in the database I removed two records
(DELETE FROM scheduled_tasks WHERE class_name = 'classes.tasks.OpenAccessNotification';
DELETE FROM scheduled_tasks WHERE class_name = 'classes.tasks.SubscriptionExpiryReminder';).

Today 2 notifications, again, appeared in our mail (instead of 4, as it was before):
Subject: 	##admin.scheduledTask.openAccessNotification## - 57f355a9dfd76 - Error
Subject: 	Subscription expiry reminder - 57f355a9504cd - Error

Now, in the database we have:
mysql> SELECT * FROM scheduled_tasks;
+--------------------------------------------------+---------------------+
| class_name                                       | last_run            |
+--------------------------------------------------+---------------------+
| classes.tasks.OpenAccessNotification             | 2016-10-04 09:09:29 |
| classes.tasks.SubscriptionExpiryReminder         | 2016-10-04 09:09:29 |
| lib.pkp.classes.task.ReviewReminder              | 2016-10-04 09:09:29 |
| plugins.generic.usageStats.UsageStatsLoader      | 2016-10-04 09:09:28 |
| plugins.importexport.crossref.CrossrefInfoSender | 2016-10-04 09:09:28 |
| plugins.importexport.datacite.DataciteInfoSender | 2016-10-04 09:09:29 |
| plugins.importexport.medra.MedraInfoSender       | 2016-10-04 09:09:29 |
+--------------------------------------------------+---------------------+

So it looks as there is still something in the code that should be removed.
I used grep to find  these tasks but I get so many lines of the search result that it is hard to figure out.
I believe you could help.

Anna

**Comment by mfelczak** [comment]
Hi @ania-appb,

Can you please clear your data cache as Site Admin -- it's likely being cached by the Acron plugin.

Cheers,
Michael

**Comment by ania-appb** [comment]
@mfelczak, today: 
1. I cleared our data cache as Site Admin 
2. I  checked  registry/scheduledTasks.xml - the fix is applied

Instead of these actions, later on we got 8 mails:

Date: 	Wed, 5 Oct 2016 10:51:32 +0200
Subject: 	##admin.scheduledTask.openAccessNotification## - 57f4bf14457c5 - Error
Subject: 	##admin.scheduledTask.openAccessNotification## - 57f4bf13cd162 - Error
Subject: 	Subscription expiry reminder - 57f4bf138f246 - Error


Date: 	Wed, 5 Oct 2016 09:31:03 +0200
Subject: 	##admin.scheduledTask.openAccessNotification## - 57f4ac37ac0cf - Error
Subject: 	##admin.scheduledTask.openAccessNotification## - 57f4ac3791c1d - Error
Subject: 	##admin.scheduledTask.openAccessNotification## - 57f4ac377d00d - Error
Subject: 	Subscription expiry reminder - 57f4ac372a9ac - Error
Subject: 	Subscription expiry reminder - 57f4ac371bfae - Error

And here is screenshot from SQL monitor:
![mysql](https://cloud.githubusercontent.com/assets/22195126/19107638/78fd7180-8aed-11e6-9bc2-4d58b57a5843.png)

What else could be done to stop receiving these emails?


Anna

**Comment by mfelczak** [comment]
Hi @ania-appb,

In your OJS3 directory, can you please open the file `cache/fc-pluginSettings-0-acronplugin.php` and check whether you have the following contents:

```
    5 =>
    array (
      'className' => 'classes.tasks.SubscriptionExpiryReminder',
      'frequency' =>
      array (
        'hour' => 24,
      ),
      'args' =>
      array (
      ),
    ),
    6 =>
    array (
      'className' => 'classes.tasks.OpenAccessNotification',
      'frequency' =>
      array (
        'hour' => 24,
      ),
      'args' =>
      array (
      ),
    ),
```

Thanks,
Michael

**Comment by mfelczak** [comment]
Hi @ania-appb,

Can you please try the following (in this order) and confirm whether this fixes the issue for you?

1) Delete Acron crontab setting from DB:
`DELETE FROM plugin_settings WHERE plugin_name = 'acronplugin' AND setting_name = 'crontab';`


2) Clear data cache as Site Admin:

> Administration => Clear Data Caches


3) Delete scheduled tasks from DB:

```
DELETE FROM scheduled_tasks WHERE class_name = 'classes.tasks.OpenAccessNotification';
DELETE FROM scheduled_tasks WHERE class_name = 'classes.tasks.SubscriptionExpiryReminder';
```


4) Disable and then enable the Acron plugin to reload new crontab settings:

> Settings => Website => Plugins => Installed Plugins => Generic Plugins => Acron Plugin
> (Unselect the checkmark to disable the plugin and then select it to enable it again)

Cheers,
Michael

**Comment by asmecher** [comment]
Committed your PR to `ojs-stable-3_0_0` and cherry-picked forward to `master`, thanks, @mfelczak -- is this one ready for closing?

**Comment by ania-appb** [comment]
Thank you very much @mfelczak, I did exactly what you described above, and this resolved our problem.

Anna

**Comment by mfelczak** [comment]
Thanks for the confirmation @ania-appb.

Thanks @asmecher, closing.

**Comment by judgej** [comment]
Just for other peope trying to resolve this problem (and for myself), what exactly are the final changes that need to be made on 3.0.0?

**Comment by mfelczak** [comment]
Hi @judgej, I've updated the instructions above (https://github.com/pkp/pkp-lib/issues/1838#issuecomment-263539251) to include a summary of all of the necessary steps to fix this issue in an OJS 3.0.0 install.
